ovos_bus_client/session.py

```diff
@@ -172,23 +173,47 @@ def from_message(message=None):
                     sess.lang = lang
             else:
                 sess = Session(lang=lang)
+                LOG.debug(f"Created new session: {sess}")
         else:
             # new session
+            LOG.warning(f"No message found, creating a new session")
             sess = Session()
+        if sess.expired():
+            # TODO: Is this the right thing to do?
+            LOG.warning(f"Resolved session expired {sess.session_id}")
+            SessionManager.sessions.pop(sess.session_id, None)
+            sess = Session(sess.lang)
         return sess
 
 
 class SessionManager:
     """ Keeps track of the current active session. """
-    default_session = None
+    default_session: Session = None
     __lock = Lock()
     sessions = {}
 
     @staticmethod
-    def reset_default_session():
+    def prune_sessions():
+        """
+        Discard any expired sessions
+        """
+        SessionManager.sessions = {sid: s for sid, s in
+                                   SessionManager.sessions.items()
+                                   if not s.expired}
+
+    @staticmethod
+    def reset_default_session() -> Session:
+        """
+        Define and return a new default_session
+        """
         with SessionManager.__lock:
             sess = Session()
             LOG.info(f"New Default Session Start: {sess.session_id}")
+            if SessionManager.default_session.session_id in \
+                    SessionManager.sessions:
+                LOG.error(f"Expired default session unexpectedly in sessions")
```

Comment:
this is not unexpected, we are reseting the session so it should always be in `SessionManager.sessions` i think ?

the pop below is good but i don't think the error is an error to be logged

Reply:
Right, when called from below it won't exist but this method could be called from elsewhere